### PR TITLE
HiSilicon D02 port

### PR DESCRIPTION
**Pull request by jforissier**
This PR adds support for the HiSilicon D02 board.

### CODE REVIEW

File: core/arch/arm/plat-d02/main.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+{
+	hi16xx_uart_init(console_base(), CONSOLE_UART_CLK_IN_HZ,
+			 CONSOLE_BAUDRATE);
+}
+
+void console_putc(int ch)
+{
+	vaddr_t base = console_base();
+
+	hi16xx_uart_putc(ch, base);
```
Comment (d3zd3z) [inline core/arch/arm/plat-d02/main.c]: I think these would normally be reversed. Usually, the CR is put out before the LF. I think most terminals are OK with it, but it is unconventional.

**Comment by d3zd3z** [comment]
Other than a minor comment,
```
Reviewed-by: David Brown <david.brown@linaro.org>
```

File: core/arch/arm/plat-d02/main.c (hunk 1 above)
Comment (vchong) [inline core/arch/arm/plat-d02/main.c]: I thought so too. All the platforms that have a `console_putc` function have this reversed. Maybe a PR later to fix them?

Comment (d3zd3z) [inline core/arch/arm/plat-d02/main.c]: Created [Fix "\n\r"](https://github.com/OP-TEE/optee_os/issues/962) issue.

Comment (jforissier) [inline core/arch/arm/plat-d02/main.c]: I'll fix that. Thanks

**Pull request merged**